### PR TITLE
fix for bug1431655: delete reencrypt routes

### PR DESCRIPTION
**Pull request by rajatchopra**
Fix for: https://bugzilla.redhat.com/show_bug.cgi?id=1431655
bug: 1431655

@lihongan Review please
@pravisankar PTAL

/cc @openshift/networking

### CODE REVIEW

File: pkg/router/f5/plugin.go [hunk 1]
```diff
@@ -510,6 +510,28 @@ func (p *F5Plugin) deleteRoute(routename string) error {
 				return err
 			}
 		}
+	} else {
```
Comment (lihongan) [inline pkg/router/f5/plugin.go]: I think should not use "else" here. That means if passthrough route existed, then no chance to delete reencrypt routes, right?

Comment (rajatchopra) [inline pkg/router/f5/plugin.go]: Yes. But aren't they mutually exclusive anyway? For the given routename.

Comment (lihongan) [inline pkg/router/f5/plugin.go]: I just saw other three type routes are checked in sequence in the function. Seems only one type route will be matched for a given routename.
LGTM. Thank you.

**Comment by knobunc** [comment]
[merge]
[test]

**Comment by knobunc** [comment]
[test] Last flaked on https://github.com/openshift/origin/issues/14043 (logs: https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/2773/ )

**Comment by rajatchopra** [comment]
[test] flake #13271

**Comment by eparis** [comment]
[test] just for funzies this time.

**Comment by knobunc** [comment]
re-[merge] last flaked on https://github.com/openshift/origin/issues/14043 (logs: https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/1172/)

**Comment by eparis** [comment]
[test]
[merge]

**Comment by knobunc** [comment]
[test] I think we flaked on the rate limiter bug https://github.com/openshift/origin/issues/12558

**Comment by knobunc** [comment]
[merge] last flaked on GCE provisioning

**Pull request merged**